src/Domain/Model/DirectDebitPayment.php

```diff
@@ -5,25 +5,19 @@
 namespace WMDE\Fundraising\PaymentContext\Domain\Model;
 
 use DateTimeImmutable;
+use WMDE\Euro\Euro;
 
-/**
- * @license GPL-2.0-or-later
- * @author Jeroen De Dauw < jeroendedauw@gmail.com >
- */
-class DirectDebitPayment implements PaymentMethod {
+class DirectDebitPayment extends Payment {
 
-	private BankData $bankData;
+	private const PAYMENT_METHOD = 'BEZ';
 
-	public function __construct( BankData $bankData ) {
-		$this->bankData = $bankData;
-	}
-
-	public function getId(): string {
-		return PaymentMethod::DIRECT_DEBIT;
-	}
+	private ?Iban $iban;
+	private ?string $bic;
```

Comment:
My "nullable properties are evil"-sense was triggered again by this, until I remember why this must be nullable. Co wondered too. My suggestion to make this more understandable would be:

- Add a comment here (e.g. `Anonymized payments have null properties`)
- make the constructor private, with nullable params
- add static creation methods, e.g. `create` and `createAnonymized`

Reply:
Would `createAnonymized` ever be called, as Doctrine will be filling in those values with reflection?

Reply:
Only in a test. That method would be just there to remind us why the props can be nullable.

Maybe a comment will suffice.